src/components/Challenges.svelte

```diff
@@ -55,94 +84,107 @@
-                    <Alert class="my-2" color="green">
-                        <span class="font-bold">Started!</span><br />
-                        Infos below.
-                    </Alert>
+{#if loading}
+    <div class="flex-1 text-center justify-center">
+        <Spinner size={'16'} />
+    </div>
+{:else if challenges.length > 0}
+    {#each sortedData as category}
+        <h1 class="text-3xl text-center mt-8">Category: {category[0].challenge_category}</h1>
+        <div class="grid grid-cols-1 lg:grid-cols-3 gap-4 lg:gap-2 place-items-center">
+            {#each category as challenge}
+                <Card
+                    class="m-4 bg-[#0000001f] dark:bg-[#0000004f] border-2 border-neutral-200 dark:border-neutral-800 backdrop-blur-3xl"
+                >
                     <div class="mb-2">
-                        <Label for="challenge-ip" class="mb-2">IP</Label>
-                        <p id="challenge-ip">{challengeResponse.details.IP}</p>
+                        <Label for="challenge-name" class="mb-2">Challenge Name</Label>
+                        <p id="challenge-name">{challenge.challenge_name}</p>
                     </div>
-                    <div class="mb-2">
-                        <Label for="challenge-port" class="mb-2">Port</Label>
-                        <p id="challenge-port">{challengeResponse.details.PORT}</p>
+                    <Accordion flush>
+                        <AccordionItem>
+                            <span slot="header">Challenge Description</span>
+                            <p class="mb-2 text-gray-500 dark:text-gray-400">{challenge.challenge_description}</p>
+                        </AccordionItem>
+                    </Accordion>
+                    <div class="mb-6 mt-6">
+                        <Label for="challenge-diff" class="mb-2">Challenge Information</Label>
+                        {#if challenge.challenge_difficulty == 'Easy'}
+                            <p>Category: {challenge.challenge_category}</p>
+                            <p id="challenge-diff" class="text-green-400">
+                                Difficulty: {challenge.challenge_difficulty}
+                            </p>
+                        {:else if challenge.challenge_difficulty == 'Medium'}
+                            <p>Category: {challenge.challenge_category}</p>
+                            <p id="challenge-diff" class="text-orange-400">
+                                Difficulty: {challenge.challenge_difficulty}
+                            </p>
+                        {:else if challenge.challenge_difficulty == 'Hard'}
+                            <p>Category: {challenge.challenge_category}</p>
+                            <p id="challenge-diff" class="text-red-400">Difficulty: {challenge.challenge_difficulty}</p>
+                        {:else}
+                            <p id="challenge-diff">Unknown</p>
+                        {/if}
```

Comment:
`challenge_difficulty` is notNull() so this can be done without a `if` since diff is always known and never unknown.

Reply:
Color check was adjusted